pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -160,51 +182,69 @@ const CryptoPolicyDialog = ({
-                       <Button id="crypto-policy-save-reboot" variant='primary' onClick={() => setPolicy(true)}>
-                           {_("Save and reboot")}
+                       {inProgress && selected === "FIPS" &&
+                           <Flex spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }} className="fips-mode-progress">
+                               <Spinner isSVG size="md" />
+                               <p>{_("Applying FIPS policy... This may take a few minutes.")}</p>
+                           </Flex>}
+                       <Button id="crypto-policy-save-reboot" variant='primary' onClick={() => setPolicy(selected, setError, setInProgress)}
+                               isDisabled={inProgress} isLoading={inProgress}
+                       >
+                           {reApply ? _("Reapply and reboot") : _("Apply and reboot")}
                        </Button>
-                       <Button id="crypto-policy-save-reboot-later" variant='secondary' onClick={() => setPolicy(false)}>
-                           {_("Save only")}
-                       </Button>
-                       <Button variant='link' onClick={close}>
+                       <Button variant='link' onClick={close} isDisabled={inProgress}>
                            {_("Cancel")}
                        </Button>
                    </>
                }
         >
             {currentCryptoPolicy && <ProfilesMenuDialogBody active_profile={currentCryptoPolicy}
                                                      change_selected={setSelected}
+                                                     isDisabled={inProgress}
                                                      profiles={policies} />}
         </Modal>
     );
 };
 
 export const CryptoPolicyStatus = () => {
-    const [requiresReboot, setRequiresReboot] = useState(false);
-    const [showShutdownModal, setShowShutDownModal] = useState(false);
+    const [currentCryptoPolicy, setCurrentCryptoPolicy] = useState(null);
+    const [fipsEnabled, setFipsEnabled] = useState(null);
+    const [showReApplyCryptoPolicy, setShowReApplyCryptoPolicy] = useState(false);
 
     useEffect(() => {
-        if (!requiresReboot) {
-            // For when we change the setting
-            cockpit.file(applyNeedRebootStamp).watch(content => setRequiresReboot(content !== null));
+        if (currentCryptoPolicy === null) {
+            cockpit.spawn(["update-crypto-policies", "--show"], { err: "message" })
+                    .then(output => setCurrentCryptoPolicy(output.trim()))
+                    .catch(setCurrentCryptoPolicy(undefined));
+            // When users run `update-crypto-policies` manually.
+            cockpit.file("/etc/crypto-policies/state/current")
+                    .watch(content => setCurrentCryptoPolicy(content ? content.trim() : undefined));
```

Comment:
These two race against each other. Chances are that `update-crypto-policies --show` will win, as it is slower. But ISTM that we should pick a Single Source of Truth.

Reply:
I ran `update-crypto-policies --show` and `--set` with a few values, and `--show` and the config file always coincide for me. So just reading the config seems ok?